jiant/models.py

```diff
@@ -1160,6 +1176,70 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _masked_lm_forward(self, batch, task, predict):
+        """
+        We currently only support RoBERTa-style dynamic masking, with the exact 
+        setup and parameters as RoBERTa. 
+        """
+        mlm_probability = 0.15
+        out = {}
+        sent_encoder = self.sent_encoder
+        tokenizer_name = self.sent_encoder._text_field_embedder.input_module
+        vocab_size = (
+            self.sent_encoder._text_field_embedder.model.embeddings.word_embeddings.num_embeddings
+        )
+        tokenizer = get_tokenizer(tokenizer_name)
+        input_key = self.sent_encoder._text_field_embedder.tokenizer_required
+        mask_idx = self.sent_encoder._text_field_embedder._mask_id
+        b_size, seq_len = batch["targs"].size()
+        inputs = batch["input"][input_key]
+        labels = batch["targs"]
+
+        probability_matrix = torch.full(labels.shape, mlm_probability, device=inputs.device)
+        padding_mask = labels.eq(0)
+        probability_matrix.masked_fill_(padding_mask, value=0.0)
+
+        masked_indices = torch.bernoulli(probability_matrix).to(
+            device=inputs.device, dtype=torch.uint8
+        )
+        tokenizer_name = self.sent_encoder._text_field_embedder.tokenizer_required
+        labels, _ = self.sent_encoder._text_field_embedder.correct_sent_indexing(
```

Comment:
It is necessary to correct_sent_indexing for labels here because the namespace for labels is padded by PAD and UNK. It was necessary to do this because the version we had was preprocessed with MosesTokenizer and there was one token that was being preprocessed wrongly.

Reply:
This might be no longer necessary once we use the non-preprocessed wikitext.